### PR TITLE
Makes Medals Cooler

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

When Medals are rewarded from one person to another, they will be announced in the corresponding departmental channel. CC has also been given a proper medal box with ALL the medals, and not just the captain's medals.

Requirements
- The medal must be one of the medals that has an assigned channel
- The medal must not have been given out before (the announcement is a once a round thing)
- The medal must be placed from one person onto another, no awarding it to yourself. (I guess you could award it to pun pun though)
- The medal announcement will only happen on station (admins can do whatever they want on CC)
- The medal's speaker can be toggled at any time by using it in hand

A list of the channels
- medal of exceptional heroism (CC only medal) - common
- medal of captaincy - no channel, this will never be announced
- medal of command - command
- medal of valor - common
- bronze heart medal - common
- Normal gold medal - common
- Normal silver medal - command
- Normal bronze medal - nothing (imagine coming in 3rd place)
- security medal - security
- medical medal - medical
- legal medal - procedure
etc...

Clicking on yourself with an accessory, will attach it instantly instead of a 4 second delay.

## Why It's Good For The Game
Medals are kinda meh for roleplay right now, no one ever real sees them unless they were there at when you got the medal, or are looking for it. This provides a cool situation where heads can award their medal to the whole department, making some cool use of some of the rarely used channels (I'm looking at you, medical).

## Images of changes

image
image

## Testing

See above, I figured out I messed some stuff up with proceedure. Normal global announcer still works regularly for

## Changelog
:cl:
tweak: Almost all medals (excluding the medal of captaincy) will make a one-time departmental announcement when put on someone other than yourself!
tweak: Clicking on yourself with an accessory, will attach it instantly instead of a 4 second delay.
/:cl:

### CODE REVIEW

**Comment by CinnamonSnowball** [comment]
How about making the speaker disable action a toggle? Just so the announcement is still usable after some funny antag decides to yoink them for a collection

**Comment by CinnamonSnowball** [comment]
How about making the speaker disable action a toggle? Just so the announcement is still usable after some funny antag decides to yoink them for a collection

**Comment by Contrabang** [comment]
> How about making the speaker disable action a toggle? Just so the announcement is still usable after some funny antag decides to yoink them for a collection

I thought about that, but I thought it would be more interesting for an antag to willingly go out of their way to disable the speakers and steal the medals (hell... that could kinda be a cool objective)

**Comment by CinnamonSnowball** [comment]
> it would be more interesting for an antag
Ehh, they're items intended for crew use, not antags, so I feel like the functionality should support crew more than the rare antag who decides to collect all the medals

**Comment by Sadhorizon** [comment]
Medal of Valor and bronze heart feel like something that should be announced on common. Have you thought about that by any chance?

**Comment by Contrabang** [comment]
> > it would be more interesting for an antag
> 
> Ehh, they're items intended for crew use, not antags, so I feel like the functionality should support crew more than the rare antag who decides to collect all the medals

Alright, I'll change it then

> Medal of valor and bronze heart feel like something that would be announced on common. Have you thought about it maybe?

Changed!

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 1]
```diff
@@ -111,6 +111,28 @@
 	new /obj/item/clothing/accessory/medal/silver/valor(src)
 	new /obj/item/clothing/accessory/medal/heart(src)
 
+/obj/item/storage/lockbox/medal/cc
+	name = "central command medal box"
+	desc = "A locked box used to store ALL medals you could ever need."
```
Comment (DGamerL) [inline code/game/objects/items/weapons/storage/lockbox.dm:116]:
```suggestion
	desc = "A locked box used to store ALL the medals you could ever need."
```

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 2]
```diff
@@ -111,6 +111,28 @@
...
+	max_combined_w_class = 30
+	storage_slots = 15
+	req_access = list(ACCESS_CENT_COMMANDER)
+
+/obj/item/storage/lockbox/medal/cc/populate_contents()
+	new /obj/item/clothing/accessory/medal/gold/heroism(src)
+	..()
+	new /obj/item/clothing/accessory/medal/gold(src)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/storage/lockbox.dm:124]: Any reason we call parent in the middle of this all? If not, could you move it to either the bottom or the top of this proc so it looks a little neater?

File: code/modules/clothing/under/accessories/accessory.dm [hunk 3]
```diff
@@ -82,13 +82,17 @@
 		if(istype(U))
 			user.visible_message("<span class='notice'>[user] is putting a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You begin to put a [src.name] on [H]'s [U.name]...</span>")
 			if(do_after(user, 40, target=H) && H.w_uniform == U)
-				user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
-				U.attackby(src, user)
+				if(U.attach_accessory(src, user, TRUE))
+					user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
```
Comment (DGamerL) [inline code/modules/clothing/under/accessories/accessory.dm:86]: Will `[src.name]` add `the` in front of it? If so, might be better to remove the `a` then

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 2 above)
Comment (Contrabang) [reply code/game/objects/items/weapons/storage/lockbox.dm:124]: CC medal, captain medals, all other medals

File: code/modules/clothing/under/accessories/accessory.dm (hunk 3 above)
Comment (Contrabang) [reply code/modules/clothing/under/accessories/accessory.dm:86]: Nope, but [src] would do that

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 4]
```diff
@@ -111,6 +111,28 @@
...
+	new /obj/item/clothing/accessory/medal/silver(src)
+	new /obj/item/clothing/accessory/medal(src)
+
```
Comment (Burzah) [inline code/game/objects/items/weapons/storage/lockbox.dm:127]:
```suggestion
```

File: code/modules/clothing/under/accessories/accessory.dm [hunk 5]
```diff
@@ -80,15 +80,22 @@
 			return 1
```
Comment (Burzah) [inline code/modules/clothing/under/accessories/accessory.dm:80]:
```suggestion
			return TRUE
```
I know you didn't touch this, but can we fix it while we are here?

File: code/modules/clothing/under/accessories/accessory.dm [hunk 6]
```diff
@@ -80,15 +80,22 @@
...
 			user.visible_message("<span class='notice'>[user] is putting a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You begin to put a [src.name] on [H]'s [U.name]...</span>")
 			if(do_after(user, 40, target=H) && H.w_uniform == U)
-				user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
-				U.attackby(src, user)
+				if(U.attach_accessory(src, user, TRUE))
+					user.visible_message("<span class='notice'>[user] puts a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You finish putting a [src.name] on [H]'s [U.name].</span>")
+					after_successful_nonself_attach(H, user)
 		else
 			to_chat(user, "[H] is not wearing anything to attach \the [src] to.")
 		return 1
```
Comment (Burzah) [inline code/modules/clothing/under/accessories/accessory.dm:93]:
```suggestion
		return TRUE
```
Same as above.

File: code/modules/clothing/under/accessories/accessory.dm [hunk 7]
```diff
@@ -80,15 +80,22 @@
 			return 1
 		var/obj/item/clothing/under/U = H.w_uniform
 		if(istype(U))
+			if(user == H)
+				U.attach_accessory(src, user, TRUE)
+				return
 			user.visible_message("<span class='notice'>[user] is putting a [src.name] on [H]'s [U.name]!</span>", "<span class='notice'>You begin to put a [src.name] on [H]'s [U.name]...</span>")
 			if(do_after(user, 40, target=H) && H.w_uniform == U)
```
Comment (Burzah) [inline code/modules/clothing/under/accessories/accessory.dm:87]:
```suggestion
			if(do_after(user, 4 SECONDS, target=H) && H.w_uniform == U)
```
I know this wasn't touched either, but can we change it while we are here?

File: code/game/objects/items/weapons/storage/lockbox.dm (hunk 4 above)
Comment (Contrabang) [reply code/game/objects/items/weapons/storage/lockbox.dm:127]:
```suggestion

	// Departmental medals
```

File: code/modules/clothing/under/accessories/accessory.dm (hunk 7 above)
Comment (Contrabang) [reply code/modules/clothing/under/accessories/accessory.dm:87]:
```suggestion
			if(do_after(user, 4 SECONDS, target = H) && H.w_uniform == U)
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**